packages/labs/src/components/select/select.tsx

```diff
@@ -149,6 +161,13 @@ export class Select<T> extends React.Component<ISelectProps<T>, ISelectState<T>>
         );
     }
 
+    public componentWillReceiveProps(nextProps: ISelectProps<T>) {
+        const nextQuery = getQuery(nextProps);
+        if (nextQuery !== undefined && this.state.query !== nextQuery) {
```

Comment:
nit: with `if` statements, compare the new thing to the existing thing:

**`nextQuery !== this.state.query`** instead of ~`this.state.query !== nextQuery`~